src/main/kotlin/com/cognifide/gradle/common/http/HttpClient.kt

```diff
@@ -75,20 +77,25 @@ open class HttpClient(private val common: CommonExtension) {
         common.prop.string("httpClient.basicPassword")?.let { set(it) }
     }
 
-    var basicCredentials: Pair<String?, String?>
-        get() = when {
-            basicUser.isPresent && basicPassword.isPresent -> (basicUser.get() to basicPassword.get())
-            else -> throw HttpException("HTTP client basic credentials are missing!")
-        }
+    var basicCredentials: Pair<String, String>?
+        get() = if (basicUser.isPresent && basicPassword.isPresent) (basicUser.get() to basicPassword.get()) else null
         set(value) {
-            basicUser.set(value.first)
-            basicPassword.set(value.second)
+            value?.let {
+                basicUser.set(value.first)
+                basicPassword.set(value.second)
+            }
         }
 
     fun fileTransferCredentials() {
         basicCredentials = common.fileTransfer.credentials
     }
 
+    var bearerToken: String? = null
+
+    fun fileTransferBearerToken() {
```

Comment:
this function looks seems to be more `val` than `fun`
maybe also refactor `fileTransferCredentials`

Reply:
done